plugins/contact/src/index.ts

```diff
@@ -69,6 +69,23 @@ export interface Channel extends AttachedDoc {
   lastMessage?: Timestamp
 }
 
+/**
+ * @public
+ */
+export enum AvatarType {
+  COLOR = 'color',
+  IMAGE = 'image',
+  GRAVATAR = 'gravatar'
+}
+
+/**
+ * @public
+ */
+export interface AvatarProvider extends Doc {
+  type: AvatarType
+  getUrl: (uri: string, size: IconSize) => string
```

Comment:
Wrongly used, please create instances of AvatarProvider inside model plugin. And have getUrl: Resource<(uri: string, size: IconSize) => string> so you will iterate via AvatarProvider and get appropriate resources if required.

Reply:
Done in https://github.com/hcengineering/anticrm/pull/2339